keepassxc-browser/content/keepassxc-browser.js

```diff
@@ -833,16 +833,28 @@ browser.runtime.onMessage.addListener(async function(req, sender) {
         } else if (req.action === 'clear_credentials') {
             kpxc.clearAllFromPage();
         } else if (req.action === 'fill_user_pass_with_specific_login') {
+            try {
```

Comment:
Would it be easier to do one try/catch inside the actual function? If it's even needed.

Reply:
inside `reconnect`? that would defeat the purpose of being able to "return" an error and also a boolean

Reply:
I mean.. why not returning false instead of throw?

Reply:
There is no sense for try catch,  remove all that. The only error you provide is 'not connected' which is obviously deduced with a boolean.

Reply:
> I mean.. why not returning false instead of throw?

because there is a difference between not connecting and there being an error connecting. [The username field checks for that and returns prematurely](https://github.com/keepassxreboot/keepassxc-browser/pull/1728/files#diff-a8d53233fe21da0571fddbcbecb85e2055b697c4e0c96328670f06c3f4deb3afR131)

Reply:
but that check also isn't really critical so I can just change `reconnect()` to return if we're connected, instead of if the connection state changed